api/git/git_test.go

```diff
@@ -15,90 +13,57 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-var bareRepoDir string
-
-func TestMain(m *testing.M) {
-	if err := testMain(m); err != nil {
-		log.Fatal(err)
-	}
-}
-
-// testMain does extra setup/teardown before/after testing.
-// The function is separated from TestMain due to necessity to call os.Exit/log.Fatal in the latter.
-func testMain(m *testing.M) error {
-	dir, err := ioutil.TempDir("", "git-repo-")
-	if err != nil {
-		return errors.Wrap(err, "failed to create a temp dir")
-	}
-	defer os.RemoveAll(dir)
-
+func setup(t *testing.T) (bareRepoDir string) {
+	dir := t.TempDir()
 	bareRepoDir = filepath.Join(dir, "test-clone.git")
 
 	file, err := os.OpenFile("./testdata/test-clone-git-repo.tar.gz", os.O_RDONLY, 0755)
 	if err != nil {
-		return errors.Wrap(err, "failed to open an archive")
+		t.Fatal(errors.Wrap(err, "failed to open an archive"))
 	}
+
 	err = archive.ExtractTarGz(file, dir)
 	if err != nil {
-		return errors.Wrapf(err, "failed to extract file from the archive to a folder %s\n", dir)
+		t.Fatal(errors.Wrapf(err, "failed to extract file from the archive to a folder %s", dir))
 	}
 
-	m.Run()
-
-	return nil
+	return
```

Comment:
use `return bareRepoDir` instead of named return parameters. the reasoning is that it's more explicit when the return has the value

Reply:
Fixed